vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java

```diff
@@ -380,10 +386,19 @@ public int readBytesAvailable() {
     @Override
     public ByteBuf readBlocking() throws IOException {
         synchronized (request.connection()) {
+            long oldOperationId = operationId;
+
             while (input1 == null && !eof) {
                 try {
                     waitingForRead = true;
-                    request.connection().wait();
+                    request.connection().wait(readTimeout);
```

Comment:
`wait` with timeout doesn't provide any info if it finished with timeout or because of a notification.
With `operationId`, we can check if anything was changed since we started waiting. Hence, gives a possibility to differentiate it.

Reply:
It would be better to just record the method start time on method entry. 

If wait() returns and there is not data ready then check if the operation has timed out or not.

wait() can return spuriously, so you can't rely on the timeout having happened when it returns.